src/ispypsa/translator/temporal_filters.py

```diff
@@ -0,0 +1,182 @@
+    """
+    if config.temporal.representative_weeks is not None:
+        snapshot = filter_snapshot_for_representative_weeks(
+            representative_weeks=config.temporal.representative_weeks,
+            snapshot=snapshot,
+            start_year=config.temporal.start_year,
+            end_year=config.temporal.end_year,
+            year_type=config.temporal.year_type,
+        )
+    return snapshot
+
+
+def filter_snapshot_for_representative_weeks(
+    representative_weeks: list[int],
+    snapshot: pd.DataFrame,
+    start_year: int,
+    end_year: int,
+    year_type: str,
+):
+    """Filters a snapshot by a list of weeks.
+
+    A snapshot is provided as a pandas DatFrame with a datetime index. The
+    snapshot may be multiple years in length. The snapshot is filtered for
+    date times that fall within the weeks defined in representative_weeks.
+    The weeks are defined as full weeks within a financial or calendar year,
+    depending on the year_type provided.
+
+    Examples:
+    >>> # Filter for first and last full weeks of each calendar year from 2020-2022
+    >>> df = pd.DataFrame(index=pd.date_range('2020-01-01', '2022-12-31', freq='H'))
+    >>> filter_snapshot_for_representative_weeks(
+    ...     representative_weeks=[0],
+    ...     snapshot=df,
+    ...     start_year=2020,
+    ...     end_year=2022,
+    ...     year_type='calendar'
+    ... ).head(3)
+    Empty DataFrame
+    Columns: []
+    Index: [2020-01-06 01:00:00, 2020-01-06 02:00:00, 2020-01-06 03:00:00]
+
+    >>> # Filter for weeks 1, 26 of financial years 2021-2022 (July 2020 - June 2022)
+    >>> df = pd.DataFrame(index=pd.date_range('2020-07-01', '2022-06-30', freq='H'))
+    >>> filter_snapshot_for_representative_weeks(
+    ...     representative_weeks=[1],
+    ...     snapshot=df,
+    ...     start_year=2021,
+    ...     end_year=2022,
+    ...     year_type='fy'
+    ... ).head(3)
+    Empty DataFrame
+    Columns: []
+    Index: [2020-07-13 01:00:00, 2020-07-13 02:00:00, 2020-07-13 03:00:00]
+
+    Args:
+        representative_weeks: list[int] of full weeks to filter for. The
+            week 0 refers to the first full week (Monday-Sunday) falling
+            with in the year.
+        snapshot: pd.DataFrame with datetime index containing the snapshot
+        start_year: int defining the start year of the snapshot (inclusive)
+        end_year: int defining the end year of the snapshot (inclusive)
+        year_type: str defining year the 'fy' for financial year or 'calendar'
+
+    Raises: ValueError if the end of week falls outside after the year end i.e.
+        for all weeks 52 or greater and for some years the week 51 (note first
+        week is 0).
+    """
+    if year_type == "fy":
+        start_year = start_year - 1
+        end_year = end_year
+        month = 7
+    else:
+        start_year = start_year
+        end_year = end_year + 1
+        month = 1
```

Comment:
Also features in snapshot.py. Could go into translator/helpers.py?

Reply:
Yep, good idea, made the change.